### PR TITLE
fix: fix publish stage duplicate bug

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description

Fixes # (issue)

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Review by zhu327 — APPROVED** [review]

File: pkg/client/apisix.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func retryEvaluator(gateway string, stage string, publishID int64, retryError *e
 				*retryError = fmt.Errorf(
 					"configuration [gateway: %s,stage: %s]  [current: %d, expected: %d] is not latest",
 					gateway, stage, resp.PublishID, publishID)
-				logging.GetLogger().Info(*retryError)
+				logging.GetLogger().Debug(*retryError)
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply pkg/client/apisix.go:146]: done

File: pkg/eventreporter/reporter.go [hunk 2]
```diff
@@ -272,7 +272,7 @@ func (r *Reporter) reportEvent(event reportEvent) {
 	}
 
 	// log event
-	logging.GetLogger().Infof("report event [name:%s,gateway:%s,stage:%s,publish_id:%s,status:%s] success",
+	logging.GetLogger().Debugf("report event [name:%s,gateway:%s,stage:%s,publish_id:%s,status:%s] success",
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply pkg/eventreporter/reporter.go:275]: done

File: pkg/registry/etcdadapter.go [hunk 3]
```diff
@@ -128,16 +128,19 @@ func (r *EtcdRegistryAdapter) ListStages(ctx context.Context) ([]StageInfo, erro
...
 			continue
 		}
-		stageMap[rm.StageInfo] = struct{}{}
+		// 去重
+		if _, ok := stageMap[rm.StageInfo.Key()]; !ok {
+			stageMap[rm.StageInfo.Key()] = rm.StageInfo
+		}
 	}
-	for stage := range stageMap {
+	for _, stage := range stageMap {
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply pkg/registry/etcdadapter.go:153]: done

**Comment by wklken** [comment]
lint 挂了

**Pull request merged**